src/EnergyPlus/CMakeLists.txt

```diff
@@ -738,11 +738,18 @@ if ((CMAKE_COMPILER_IS_GNUCXX AND CMAKE_CXX_COMPILER_VERSION VERSION_LESS 9.1) O
     ( ("${CMAKE_CXX_COMPILER_ID}" STREQUAL "Clang" OR "${CMAKE_CXX_COMPILER_ID}" STREQUAL "AppleClang") AND (CMAKE_CXX_COMPILER_VERSION VERSION_LESS 9.0)))
   target_link_libraries(energyplusparser PRIVATE stdc++fs)
 endif()
+if(ENABLE_UNITY)
+  set_target_properties(energyplusparser PROPERTIES UNITY_BUILD ON)
+endif()
```

Comment:
The other question I have is whether that is something that should be done like this, or doing it the opposite way.

See the "Note" section here: https://cmake.org/cmake/help/latest/prop_tgt/UNITY_BUILD.html , emphasis is mine:

> Projects **should not directly set the UNITY_BUILD property** or its associated [CMAKE_UNITY_BUILD](https://cmake.org/cmake/help/latest/variable/CMAKE_UNITY_BUILD.html#variable:CMAKE_UNITY_BUILD) variable to true. Depending on the capabilities of the build machine and compiler used, it might or might not be appropriate to enable unity builds. Therefore, **this feature should be under developer control, which would normally be through the developer choosing whether or not to set the [CMAKE_UNITY_BUILD](https://cmake.org/cmake/help/latest/variable/CMAKE_UNITY_BUILD.html#variable:CMAKE_UNITY_BUILD) variable on the [cmake(1)](https://cmake.org/cmake/help/latest/manual/cmake.1.html#manual:cmake(1)) command line** or some other equivalent method. However, **it IS recommended to set the UNITY_BUILD target property to false if it is known that enabling unity builds for the target can lead to problems**.

By "the opposite", I mean remove these changes that turn it on, set the test targets and co to `OFF` instead if they violate ODR, and pass `-DCMAKE_UNITY_BUILD:BOOL=ON` instead.

Disclaimer: I'm not familiar with unity builds.

Reply:
I hadn't read in detail this part of the doc, thanks for pointing it out!

In any case this option has to be opt-in, the question is whether it should be with `-DENABLE_UNITY=ON` or `-DCMAKE_UNITY_BUILD=ON`. Here I merely replicated the way we use it at my company - which is not necessarily the right way.

Using the latter would be more standard for anybody used to CMake (standard flag being used), and would prevent misbehaviour for projects that do not support it in case someone sets `-DCMAKE_UNITY_BUILD=ON` (currently it would turn it on for all targets, including the ones that do not support it).

I don't mind giving it a try, but many third-party targets do not compile when setting it, so that will result in more changes.

Reply:
Ok, I think that makes sense and I'm satisfied with the current implementation.